pkg/graph/resolvers/system_intake.go

```diff
@@ -370,21 +369,52 @@ func GetMySystemIntakes(ctx context.Context, store *storage.Store) ([]*models.Sy
 	return store.GetMySystemIntakes(ctx)
 }
 
-func GetRequesterUpdateEmailData(ctx context.Context, store *storage.Store) ([]*models.RequesterUpdateEmailData, error) {
+func GetRequesterUpdateEmailData(
+	ctx context.Context,
+	store *storage.Store,
+	fetchUserInfos func(context.Context, []string) ([]*models.UserInfo, error),
+) ([]*models.RequesterUpdateEmailData, error) {
+
 	// first, get data from store
 	data, err := store.GetRequesterUpdateEmailData(ctx)
 	if err != nil {
 		return nil, err
 	}
 
+	var euaIDs []string
+	for _, item := range data {
```

Comment:
NIT: 
I think this is fine for the PR, but I suggest it might be easier for readability to use the `samber.lo` package for some of these mappings and data manipulations. Since we use `lo` throughout the codebase, you just need to read the function name to know what's going on instead of reading the loop.